src/component-library/CTA/CTA.tsx

```diff
@@ -1,44 +1,49 @@
 import { ButtonHTMLAttributes, forwardRef } from 'react';
 
-import { CTAVariants, Sizes } from '../utils/prop-types';
-import { OutlinedCTA, PrimaryCTA, SecondaryCTA } from './CTA.style';
+import { LoadingSpinner } from '../LoadingSpinner';
+import { Sizes, Variants } from '../utils/prop-types';
+import { BaseCTA, BaseCTAProps } from './BaseCTA';
+import { LoadingWrapper } from './CTA.style';
+
+const loadingSizes: Record<Sizes, number> = {
+  small: 16,
+  medium: 18,
+  large: 20
+};
 
 type Props = {
-  variant?: CTAVariants;
+  variant?: Variants;
   fullWidth?: boolean;
   size?: Sizes;
+  loading?: boolean;
 };
 
 type NativeAttrs = Omit<ButtonHTMLAttributes<unknown>, keyof Props>;
 
-type CTAProps = Props & NativeAttrs;
+type InheritAttrs = Omit<BaseCTAProps, keyof Props & NativeAttrs>;
+
+type CTAProps = Props & InheritAttrs & NativeAttrs;
 
 const CTA = forwardRef<HTMLButtonElement, CTAProps>(
-  (
-    { disabled = false, variant, fullWidth = false, size = 'medium', onClick, className, children, ...rest },
-    ref
-  ): JSX.Element => {
-    const props = {
-      as: 'button' as keyof JSX.IntrinsicElements,
-      disabled,
-      $fullWidth: fullWidth,
-      ref,
-      onClick,
-      className,
-      children,
-      $size: size,
-      ...rest
-    };
-
-    switch (variant) {
-      default:
-      case 'primary':
-        return <PrimaryCTA {...props} />;
-      case 'secondary':
-        return <SecondaryCTA {...props} />;
-      case 'outlined':
-        return <OutlinedCTA {...props} />;
-    }
+  ({ children, loading, disabled, variant, fullWidth, size = 'medium', ...props }, ref): JSX.Element => {
+    const isDisabled = disabled || loading;
+
+    return (
+      <BaseCTA ref={ref} as='button' disabled={isDisabled} fullWidth={fullWidth} {...props}>
```

Comment:
Minor bug here: `size` and `variant` properties are not passed into `BaseCTA` component and therefore the CTA always uses default values even if we pass these props.